consensus/istanbul/core/roundchange.go

```diff
@@ -122,21 +128,28 @@ func (c *core) handleRoundChange(msg *message, src istanbul.Validator) error {
 
 	if num == numStartNewRound && (c.waitingForRoundChange || cv.Round.Cmp(roundView.Round) < 0) {
 		// We've received enough ROUND CHANGE messages, start a new round immediately.
-		logger.Warn("[RC] Received 2f+1 Round Change Messages. Starting new round")
+		logger.Warn("[RC] Prepare messages received before startNewRound", "round", cv.Round.String(),
+			"len(prepares)", c.current.Prepares.Size(), "messages", c.current.Prepares.GetMessages())
+		logger.Warn("[RC] Commit messages received before startNewRound", "round", cv.Round.String(),
+			"len(commits)", c.current.Commits.Size(), "messages", c.current.Commits.GetMessages())
+		logger.Warn("[RC] Received 2f+1 Round Change Messages. Starting new round",
+			"currentRound", cv.Round.String(), "newRound", roundView.Round.String())
```

Comment:
How about merging logs into one?

Reply:
`messages` could contain tens of node addresses. Isn't it too long if they are combined?

Reply:
If so, it doesn't make much difference 😅 LGTM